File: admin-web/src/api/operation-log.ts
import { http } from './http'

export interface OperationLogVO {
  id: number
  operatorId: string
  operatorName: string
  operationType: string
  targetType: string
  targetId: number
  targetName: string
  detail: string
  createdAt: string
}

export interface OperationLogQueryParams {
  page?: number
  size?: number
  operationType?: string
  targetType?: string
  operatorId?: string
  startDate?: string
  endDate?: string
}

export function listOperationLogs(params: OperationLogQueryParams = {}) {
  return http.get('/admin/operation-logs', { params })
}

File: backend/src/main/java/com/marketing/system/domain/entity/OperationLog.java
package com.marketing.system.domain.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("operation_log")
public class OperationLog {
    @TableId(type = IdType.AUTO)
    private Long id;
    private String operatorId;
    private String operatorName;
    private String operationType;
    private String targetType;
    private Long targetId;
    private String targetName;
    private String detail;
    private LocalDateTime createdAt;
}
